### PR TITLE
Improve type safety

### PR DESCRIPTION
**Pull request by vogelsgesang**
This pull request contains a few step towards improved type safety.
In particular, the `standalone-server` is now completely annotated with TypeScript types.
Also, the `query-graphs` core library already contains a few type annotations, however for the core library there still remains a lot of work...

### CODE REVIEW

File: query-graphs/src/common.ts [hunk 1]
```diff
@@ -88,6 +91,28 @@ export function toString(d) {
     return undefined;
 }
 
+export function json2stringmap(json: string): Map<string, string> {
```
Comment (rickcole) [inline query-graphs/src/common.ts:94]: Function name casing?

File: query-graphs/src/tableau.ts [hunk 2]
```diff
@@ -485,6 +485,25 @@ function collapseNodes(treeData, graphCollapse) {
     }
 }
 
+// Color graph per federated connections
+export function colorFederated(treeData) {
```
Comment (rickcole) [inline query-graphs/src/tableau.ts:489]: Export no longer necessary?

File: query-graphs/src/tree-description.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+export type GraphOrientation = "left-to-right" | "top-to-bottom" | "right-to-left" | "bottom-to-top";
+
+export interface TreeNode {
```
Comment (rickcole) [inline query-graphs/src/tree-description.ts:3]: Nice!

File: query-graphs/src/tableau.ts (hunk 2 above)
Comment (vogelsgesang) [reply query-graphs/src/tableau.ts:489]: fixed

File: query-graphs/src/common.ts (hunk 1 above)
Comment (vogelsgesang) [reply query-graphs/src/common.ts:94]: renamed to jsonToStringMap

**Pull request merged**